packages/testcontainers/src/containers/NativeChainContainer.ts

```diff
@@ -0,0 +1,671 @@
+    ],
+    testnet: [
+      '-testnet=1'
+    ],
+    regtest: [
+      '-regtest=1',
+      '-jellyfish_regtest=1',
+      '-txnotokens=0',
+      '-logtimemicros',
+      '-txindex=1',
+      '-acindex=1',
+      '-amkheight=0',
+      '-bayfrontheight=1',
+      '-bayfrontgardensheight=2',
+      '-clarkequayheight=3',
+      '-dakotaheight=4',
+      '-dakotacrescentheight=5',
+      '-eunosheight=6',
+      '-eunospayaheight=7',
+      '-fortcanningheight=8',
+      '-fortcanningmuseumheight=9',
+      '-fortcanninghillheight=10',
+      '-fortcanningroadheight=11',
+      '-fortcanningcrunchheight=12',
+      '-fortcanningspringheight=13',
+      '-fortcanninggreatworldheight=14',
+      '-fortcanningepilogueheight=15'
+    ],
+    mainnet: [],
+    devnet: [],
+    masternode: [
+      '-dummypos=0',
+      '-spv=1',
+      '-anchorquorum=2'
+    ]
+  }
+
+  /**
+   * Convenience Cmd builder
+   */
+  protected generateCmd (): string[] {
+    const { defaultCmd } = NativeChainContainer
+
+    const authCmd = [
+      `-rpcuser=${this.rpcUser}`,
+      `-rpcpassword=${this.rpcPassword}`
+    ]
+
+    const masterNodeOperator = (this.masterNodeKey != null)
+      ? [`-masternode_operator=${this.masterNodeKey.operator.address}`]
+      : []
+
+    return [
+      ...defaultCmd.prepend,
+      ...authCmd,
+      ...defaultCmd[this.blockchainNetwork.name],
+      ...this.hasPreconfiguredMasternode ? defaultCmd.masternode : [],
+      ...masterNodeOperator,
+      ...this.addedCmds
+    ]
+  }
+
+  private readonly addedCmds: string[] = []
+  public addCmd (newCmd: string): this {
+    this.addedCmds.push(newCmd)
+    return this
+  }
+
+  /**
+   * Create container and start it immediately waiting for NativeChain to be ready
+   */
+  public async start (): Promise<StartedNativeChainContainer> {
+    this.withExposedPorts(...(this.hasExposedPorts ? this.ports : Object.values(this.blockchainNetwork.ports)))
+      .withName(this.name ?? this.generateName())
+      .withCommand(this.command.length > 0 ? this.command.concat(this.addedCmds) : this.generateCmd())
+
+    const {
+      rpcUser,
+      rpcPassword,
+      blockchainNetwork,
+      masterNodeKey
+    } = this
+    const startedContainer = new StartedNativeChainContainer(
+      await super.start(),
+      { rpcUser, rpcPassword, blockchainNetwork, masterNodeKey })
+
+    if (masterNodeKey != null) {
+      await startedContainer.call('importprivkey', [masterNodeKey.operator.privKey, 'operator', true])
+      await startedContainer.call('importprivkey', [masterNodeKey.owner.privKey, 'owner', true])
+    }
+
+    return startedContainer
+  }
+}
+
+export interface StartedContainerConfig {
+  rpcUser: string
+  rpcPassword: string
+  blockchainNetwork: BlockchainNetwork
+  masterNodeKey?: MasterNodeKey
+}
+
+export class StartedNativeChainContainer extends AbstractStartedContainer {
+  private assumedSpvHeight: number = 0
+  static SPV_EXPIRATION = 10
+  protected rpcUrl: string
+
```

Comment:
I'm thinking if it's cleaner to keep `StartedNativeChainContainer` simple by moving container `Rpc` and `WaitFor` utilities other classes.

```suggestion

  rpc = new NativeChainRpc(this)
  waitFor = new NativeChainWaitFor(rpc)
```